src/encoder/encoderwave.cpp

```diff
@@ -0,0 +1,214 @@
+        m_sfInfo.format = SF_FORMAT_AIFF;
+    }
+    else {
+        qWarning() << "Unexpected Format when setting EncoderWave: " << format.internalName << ". Reverting to wav";
+        // Other possibly interesting formats 
+        // There is a n option for RF64 to automatically downgrade to RIFF WAV if less than 4GB using an
+        // sf_command, so it could be interesting to use it in place of FORMAT_WAVE.
+        // SF_FORMAT_W64          = 0x0B0000,     /* Sonic Foundry's 64 bit RIFF/WAV */
+        // SF_FORMAT_RF64         = 0x220000,     /* RF64 WAV file */
+
+        // I guess this one is WAVEFORMATEXTENSIBLE, not WAVEFORMATEX.
+        // Not really useful for us since it's mostly for multichannel setups.
+        // SF_FORMAT_WAVEX        = 0x130000,     /* MS WAVE with WAVEFORMATEX */
+
+        // SF_FORMAT_CAF          = 0x180000,     /* Core Audio File format */
+    }
+    int radio = settings.getSelectedOption(EncoderWaveSettings::BITS_GROUP);
+    switch(radio) {
+        case 0:
+            m_sfInfo.format |= SF_FORMAT_PCM_16;
+            break;
+        case 1:
+            m_sfInfo.format |= SF_FORMAT_PCM_24;
+            break;
+        case 2:
+            m_sfInfo.format |= SF_FORMAT_FLOAT;
+            break;
+        default:
+            m_sfInfo.format = SF_FORMAT_WAV | SF_FORMAT_PCM_16;
+            qWarning() << " Unexpected radio index on EncoderWave: " 
+                    << radio << ". reverting to PCM 16bits";
+            break;
+    }
+}
+
+// call sendPackages() or write() after 'flush()' as outlined in enginebroadcast.cpp
+void EncoderWave::flush() {
+    sf_write_sync(m_pSndfile);
+}
+
+
+void EncoderWave::encodeBuffer(const CSAMPLE *pBuffer, const int iBufferSize) {
+    sf_write_float(m_pSndfile, pBuffer, iBufferSize);
+}
+
+/* Originally called from enginebroadcast.cpp to update metadata information
+ * when streaming, however, this causes pops
+ *
+ * Currently this method is used before init() once to save artist, title and album
+*/
+void EncoderWave::updateMetaData(const QString& artist, const QString& title, const QString& album) {
+    m_metaDataTitle = title;
+    m_metaDataArtist = artist;
+    m_metaDataAlbum = album;
+}
+
+void EncoderWave::initStream() {
+
+    // Tell the encoder to automatically convert float input range to the correct output range.
+    sf_command(m_pSndfile, SFC_SET_NORM_FLOAT, NULL, SF_TRUE);
+    // Tell the encoder that, when converting to integer formats, clip
+    // automatically the values that go outside of the allowed range.
+    // Warning! Depending on how libsndfile is compiled autoclip may not work.
+    // Ensure CPU_CLIPS_NEGATIVE and CPU_CLIPS_POSITIVE is setup properly in the build.
+    sf_command(m_pSndfile, SFC_SET_CLIPPING, NULL, SF_TRUE) ;
+
+    // Set meta data. Latin1 since WAVE does not officially support UTF8.
+    int ret = sf_set_string(m_pSndfile, SF_STR_TITLE, m_metaDataTitle.toLatin1().constData());
```

Comment:
According to  http://www.johnloomis.org/cpe102/asgn/asgn1/riff.html

> LIST chunks of type INFO are optional in current RIFF forms, but their use is recommended. The LIST chunks' subchunks can store much more information about the file than is available from the filename and date stamp. These LIST subchunks share a common format: Each contains one ASCIIZ (NULL terminated) string. 

This means we have to use toAscii() to be standard conform. 

Could you put this quote above to the source? 

Unfortunately I think this will replace all non ASCII character by ? 

I have checked with GStreamer based Totem player. It is able to detect the encodeing of the non ASCII characters and Displays Latin1 and utf8 alternative correct. 

What to do here? 

We may remove non ASCII characters from the edit boxes of pop up a warning in case of WAV/AIFF to avoid surprised. If we want to go an extra mile, we may add a checkbox "Utf8 encoding (not RIFF standard)"